test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,266 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from haystack import Document
+from haystack.components.preprocessors.markdown_header_splitter import MarkdownHeaderSplitter
+
+
+# Fixtures
+@pytest.fixture
+def sample_text():
+    return (
+        "# Header 1\n"
+        "Content under header 1.\n"
+        "## Header 1.1\n"
+        "### Subheader 1.1.1\n"
+        "Content under sub-header 1.1.1\n"
+        "## Header 1.2\n"
+        "### Subheader 1.2.1\n"
+        "Content under header 1.2.1.\n"
+        "### Subheader 1.2.2\n"
+        "Content under header 1.2.2.\n"
+        "### Subheader 1.2.3\n"
+        "Content under header 1.2.3."
+    )
+
+
+# Basic splitting and structure
+def test_basic_split(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Should split into all headers with content
+    headers = [doc.meta["header"] for doc in split_docs]
+    assert "Header 1" in headers
+    assert "Subheader 1.1.1" in headers
+    assert "Subheader 1.2.1" in headers
+    assert "Subheader 1.2.2" in headers
+    assert "Subheader 1.2.3" in headers
+
+    # Check that content is present and correct
+    for doc in split_docs:
+        assert doc.content.startswith("#") or doc.content.startswith("##") or doc.content.startswith("###")
+        assert doc.meta.get("header") is not None
+
+
+def test_split_parentheaders(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=sample_text), Document(content="# H1\n## H2\n### H3\nContent")]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    # Check parentheaders for both a deep subheader and a simple one
+    subheader_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.2")
+    assert "Header 1" in subheader_doc.meta["parentheaders"]
+    assert "Header 1.2" in subheader_doc.meta["parentheaders"]
+    h3_doc = next((doc for doc in split_docs if doc.meta["header"] == "H3"), None)
+    if h3_doc:
+        assert h3_doc.meta["parentheaders"] == ["H1", "H2"]
```

Comment:
Shouldn't `h3_doc` always exist in this test? If so let's drop the `if h3_doc` so we can be sure we are testing the assert